### PR TITLE
KUBE-303: escape organization ID in Ops Manager API paths

### PR DESCRIPTION
**Pull request by mircea-cosbuc**
# Summary

The operator reads the "orgId" value verbatim from the user-editable project ConfigMap and splices it into the Ops Manager REST path via fmt.Sprintf with no url.PathEscape. An attacker who can edit the ConfigMap therefore controls the entire URI suffix after /orgs/ and can inject query strings, fragments, extra path segments or ".." traversal into the operator's digest-authenticated GET, redirecting it to arbitrary Ops Manager endpoints (agent API keys, automation secrets).

Fix: url.PathEscape the orgID at the three sinks in omclient.go (ReadOrganization, ReadProjectsInOrganizationByName, ReadProjectsInOrganization) so the value stays a single URL-escaped path segment.

Tests:
- controllers/om/omclient_test.go: deterministic sink-level regression guards asserting the escaped request-target for /, ?, #, and ".." payloads (fail pre-fix, pass post-fix).
- tests/opsmanager/om_ops_manager_ssrf_orgid.py: full e2e replication against a real Ops Manager (marker e2e_om_ops_manager_ssrf_orgid), registered on the kind-only OM task groups.

## Proof of Work

[EVG Patch Link](https://spruce.corp.mongodb.com/version/6a6b64ed7998e000072a8b1f/)

## Checklist

- [x] Have you linked a jira ticket and/or is the ticket in the title?
- [x] Have you checked whether your jira ticket required DOCSP changes?
- [x] Have you added changelog file?
    - use `skip-changelog` label if not needed
    - refer to [Changelog files and Release Notes](https://github.com/mongodb/mongodb-kubernetes/blob/master/CONTRIBUTING.md#changelog-files-and-release-notes) section in CONTRIBUTING.md for more details

### CODE REVIEW

**Review by filipcirtog — APPROVED** [review]
lgtm! IMO we could remove the test or at least run it only at merge time

File: controllers/om/omclient_test.go [hunk 1]
```diff
@@ -241,3 +243,147 @@ func automationConfig(groupId string, responses ...automationConfigResponse) (ha
 	}
 	return handle, counters
 }
+
+// ============================ SECBUG-4043 ============================
+// SECBUG-4043: the organization ID is read verbatim from the user-editable project
```
Comment (nammn) [inline controllers/om/omclient_test.go]: is this information we should share here?

File: docker/mongodb-kubernetes-tests/tests/opsmanager/om_ops_manager_ssrf_orgid.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+from kubetester.kubetester import KubernetesTester
+from kubetester.kubetester import fixture as yaml_fixture
+from kubetester.mongodb import MongoDB
+from kubetester.opsmanager import MongoDBOpsManager
+from kubetester.phase import Phase
+from pytest import fixture, mark
+from tests.conftest import is_multi_cluster
+from tests.opsmanager.withMonitoredAppDB.conftest import enable_multi_cluster_deployment
+
+# SECBUG-4043: end-to-end replication and regression guard for the "orgId" SSRF.
```
Comment (nammn) [inline docker/mongodb-kubernetes-tests/tests/opsmanager/om_ops_manager_ssrf_orgid.py]: do we need an e2e test, would a unit test suffice, given its boundary is the omclient?

**Review by nammn — COMMENTED** [review]
2 q

File: docker/mongodb-kubernetes-tests/tests/opsmanager/om_ops_manager_ssrf_orgid.py (hunk 2 above)
Comment (mircea-cosbuc) [reply docker/mongodb-kubernetes-tests/tests/opsmanager/om_ops_manager_ssrf_orgid.py]: I'm tempted to say we don't need an E2E test. I will remove it.

File: controllers/om/omclient_test.go (hunk 1 above)
Comment (mircea-cosbuc) [reply controllers/om/omclient_test.go]: Nope, cleaning up.

File: changelog/20260730_fix_escape_the_organization_id_read_from_the_project.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+kind: fix
+date: 2026-07-30
+---
+
+* Escape the organization ID read from the project ConfigMap when building Ops Manager API request paths, preventing request forgery to arbitrary Ops Manager endpoints (SECBUG-4043).
```
Comment (nammn) [inline changelog/20260730_fix_escape_the_organization_id_read_from_the_project.md]: should we link the fix here?

**Review by nammn — APPROVED** [review]
one comment ptal

File: changelog/20260730_fix_escape_the_organization_id_read_from_the_project.md (hunk 3 above)
Comment (mircea-cosbuc) [reply changelog/20260730_fix_escape_the_organization_id_read_from_the_project.md]: You mean the ticket, no. Thanks for catching this.
